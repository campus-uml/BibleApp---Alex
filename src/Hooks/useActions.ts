export const useActions = () => {
  const copyToClipboard = (text: string) => {
    navigator.clipboard.writeText(text).then(() => {
      alert("Texto copiado al portapapeles");
    });
  };

  const shareToWhatsApp = (text: string) => {
    const whatsappUrl = `https://wa.me/?text=${encodeURIComponent(text)}`;
    window.open(whatsappUrl, "_blank");
  };

  return {
    copyToClipboard,
    shareToWhatsApp,
  };
};
