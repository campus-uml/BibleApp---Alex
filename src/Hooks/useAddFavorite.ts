import { Verse } from "@/types";
import { useState, useEffect } from "react";

export const useAddFavorite = () => {
  const [favorites, setFavorites] = useState<Verse[]>([
    {
      id: "1",
      text: "En el principio Dios creo los cielos y la tierra.",
      reference: "Genesis 1:1",
    },
  ]);

  useEffect(() => {
    localStorage.setItem("favorites", JSON.stringify(favorites));
  }, [favorites]);

  const addFavorite = (verseId: string) => {
    setFavorites((prev) => {
      if (prev.some((fav) => fav.id === verseId)) {
        return prev;
      }

      const verse = getVerseById(verseId);
      return [...prev, verse];
    });
  };

  const removeFavorite = (verseId: string) => {
    setFavorites((prev) => prev.filter((verse) => verse.id !== verseId));
  };

  return {
    favorites,
    addFavorite,
    removeFavorite,
  };
};

const getVerseById= (verseId: string): Verse => {
  return {
    id: verseId,
    text: "En el principio Dios creo los cielos y la tierra.",
    reference: "Genesis 1:1",
  };
}