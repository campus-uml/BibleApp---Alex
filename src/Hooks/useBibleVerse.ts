import { useEffect, useState } from "react";
import { getBibles, getBiblesChapters } from "../services/getData";
import { BibleBooks, BibleChapter } from "../types/index";

export const useBibleVerse = () => {
  const [bibleVerse, setBibleVerse] = useState<BibleBooks[]>([]);
  const [bibleVerseChapters, setBibleVerseChapters] = useState<BibleChapter[]>(
    []
  );
  const [seletedBook, setSelectedBook] = useState<string | null>(null);

  useEffect(() => {
    fetchData();
    if (seletedBook) {
      fetchChapters(seletedBook);
    }
  }, [seletedBook]);

  const fetchData = async () => {
    try {
      const response = await getBibles();
      if (response && response.data) {
        setBibleVerse(response.data);
      } else {
        console.error("Unexpected API response structure:", response);
      }
    } catch (error) {
      console.error(error);
    }
  };

  const fetchChapters = async (bookId: string) => {
    try {
      const response = await getBiblesChapters(bookId);
      if (response && response.data) {
        setBibleVerseChapters(response.data);
      } else {
        console.error("Unexpected API response structure:", response);
      }
    } catch (error) {
      console.error(error);
    }
  };

  const handleBook = (bookId: string) => {
    setSelectedBook(bookId);
  };

  return {
    handleBook,
    bibleVerse,
    bibleVerseChapters,
    seletedBook,
  };
};
