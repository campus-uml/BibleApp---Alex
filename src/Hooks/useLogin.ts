import { useState } from "react";
import { useNavigate } from "react-router-dom";
import { supabase } from "@/constants/api";
import { AuthError } from "@supabase/supabase-js";

const useLogin = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [fullName, setFullName] = useState(""); 
  const [loading, setLoading] = useState(false);
  const navigate = useNavigate();

  const handleEmailLogin = async (e: React.FormEvent) => {
    e.preventDefault();
    setLoading(true);
    const { error } = await supabase.auth.signInWithPassword({
      email,
      password,
    });
    if (error) {
      alert(error.message);
    } else {
      navigate("/home");
    }
    setLoading(false);
  };

  const signUpNewUser = async (
    email: string,
    password: string,
    fullName: string
  ) => {
    setLoading(true);
    try {
      const { data, error } = await supabase.auth.signUp({
        email,
        password,
        options: {
          emailRedirectTo: `${window.location.origin}/home`,
        },
      });

      if (error) throw error;

      if (data?.user) {
        const { error: profileError } = await supabase.from("profiles").upsert({
          id: data.user.id,
          full_name: fullName,

        });
        setFullName(fullName);


        if (profileError) {
          throw profileError;
        }

        return {
          success: true,
          message: "Registro exitoso. Por favor, verifica tu email.",
        };
      } else {
        return {
          success: false,
          message: "No se pudo crear el usuario. Por favor, intenta de nuevo.",
        };
      }
    } catch (error) {
      console.error("Error signing up:", error);
      let errorMessage = "Error al registrarse. Por favor, intenta de nuevo.";

      if (error instanceof AuthError) {
        errorMessage = `Error ${error.status}: ${error.message}`;
        console.error("Detailed error:", JSON.stringify(error, null, 2));
      }

      return { success: false, message: errorMessage };
    } finally {
      setLoading(false);
    }
  };

  const handleGithubLogin = async () => {
    const { error } = await supabase.auth.signInWithOAuth({
      provider: "github",
      options: {
        redirectTo: `${window.location.origin}/home`,
      },
    });
    if (error) {
      alert("Error logging in with GitHub");
    }
  };

  const handleGoogleLogin = async () => {
    console.log("Google login");
    const { error } = await supabase.auth.signInWithOAuth({
      provider: "google",
      options: {
        redirectTo: `${window.location.origin}/home`,
      },
    });
    if (error) {
      alert("Error logging in with Google");
    }
  };

  //no se usara por ahora
  const handleAzureLogin = async () => {
    const { error } = await supabase.auth.signInWithOAuth({
      provider: "azure",
      options: {
        redirectTo: `${window.location.origin}/home`,
      },
    });
    if (error) {
      alert("Error logging in with Microsoft");
    }
  };

  return {
    email,
    setEmail,
    password,
    setPassword,
    loading,
    handleEmailLogin,
    handleGithubLogin,
    handleGoogleLogin,
    handleAzureLogin,
    signUpNewUser,
    fullName,
    
  };
};

export default useLogin;
