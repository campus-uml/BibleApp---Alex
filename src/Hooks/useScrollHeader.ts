import { useState, useEffect } from "react";

export function useScrollHeader() {
  const [isVisible, setIsVisible] = useState(true);
  const [lastScrollY, setLastScrollY] = useState(0);
  const scrollThreshold = 10; 

  useEffect(() => {
    const controlHeader = () => {
      const currentScrollY = window.scrollY;

      if (currentScrollY > lastScrollY + scrollThreshold) {
        setIsVisible(false);
      } else if (currentScrollY < lastScrollY - scrollThreshold) {
        setIsVisible(true);
      }

      setLastScrollY(currentScrollY);
    };

    window.addEventListener("scroll", controlHeader);

    return () => {
      window.removeEventListener("scroll", controlHeader);
    };
  }, [lastScrollY]); 

  return isVisible;
}
