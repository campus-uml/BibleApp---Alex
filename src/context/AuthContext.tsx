import { createContext, useContext, useEffect, useState } from "react";
import { supabase } from "..//constants/api";
import { User } from "@/types";

interface AuthContextType {
  user: User | null;
  logout: () => void;
}

const AuthContext = createContext<AuthContextType | null>(null);

export const AuthProvider = ({ children }: { children: React.ReactNode }) => {
  const [user, setUser] = useState<User | null>(null);

  useEffect(() => {
    const fetchUser = async () => {
      const { data } = await supabase.auth.getUser();
      if (data.user) setUser(data.user);
    };

    fetchUser();

    const { data: listener } = supabase.auth.onAuthStateChange((_event, session) => {
      setUser(session?.user || null);
    });

    return () => {
      listener?.subscription.unsubscribe();
    };
  }, []);

  const logout = async () => {
    await supabase.auth.signOut();
    setUser(null);
  };

  return (
    <AuthContext.Provider value={{ user, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

// eslint-disable-next-line react-refresh/only-export-components
export const useAuth = () => {
  return useContext(AuthContext);
};
