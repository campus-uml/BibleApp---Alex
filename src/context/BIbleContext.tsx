import React, {
  createContext,
  useContext,
  useEffect,
  useRef,
  useState,
} from "react";
import {
  getBibles,
  getChapters,
  loadChapterVersesFromAPI,
  searchBible,
} from "../services/getData";
import { BibleBooks, Chapter, Verse } from "../types/index";
import { useSidebar } from "@/components/ui/sidebar";

const formatVerseText = (text: string) => {
  return text.toLowerCase().replace(/^([a-z])/, (match) => match.toUpperCase());
};

const isTitleOrHeading = (text: string) => {
  const titleKeywords = ["Capítulo", "Libro", "Título"];
  return titleKeywords.some((keyword) => text.includes(keyword));
};

interface BibleContextProps {
  bibleVerse: BibleBooks[];
  selectedBook: BibleBooks["id"];
  bibleVerseChapters: Chapter[];
  chapterVerses: Verse[];
  scrollAreaRef: React.RefObject<HTMLDivElement>;
  handleBook: (bookId: string) => void;
  loadChapterVerses: (chapterId: string) => void;
  searchBibleVerse: (query: string) => void;
  toggleSidebar: () => void;
  
}

const BibleContext = createContext<BibleContextProps | undefined>(undefined);

interface BibleProviderProps {
  children: React.ReactNode;
}

export const BibleProvider: React.FC<BibleProviderProps> = ({ children }) => {
  const [bibleVerse, setBibleVerse] = useState<BibleBooks[]>([]);
  const [selectedBook, setSelectedBook] = useState<BibleBooks["id"]>("GEN");
  const [bibleVerseChapters, setBibleVerseChapters] = useState<Chapter[]>([]);
  const [chapterVerses, setChapterVerses] = useState<Verse[]>([]);
  const scrollAreaRef = useRef<HTMLDivElement>(null);
  const { toggleSidebar } = useSidebar();

  useEffect(() => {
    fetchData();
  }, []);

  useEffect(() => {
    if (selectedBook) {
      loadChapters(selectedBook);
    }
  }, [selectedBook]);

  useEffect(() => {
    if (bibleVerseChapters.length > 0) {
      loadChapterVerses(bibleVerseChapters[0].id);
    }
  }, [bibleVerseChapters]);

  const loadChapterVerses = async (chapterId: string) => {
    try {
      const response = await loadChapterVersesFromAPI(chapterId);
      if (!response.data) {
        console.error(
          "Error: No se encontró contenido en los versículos",
          response
        );
        return;
      }

      const content = response.data.content.replace(/<\/?[^>]+(>|$)/g, "");
      const versePattern = /(\d+)([^0-9]+)/g;
      const versesArray: Verse[] = [];
      let match;

      while ((match = versePattern.exec(content)) !== null) {
        const verseNumber = match[1];
        const verseText = formatVerseText(match[2].trim());

        if (!isTitleOrHeading(verseText)) {
          versesArray.push({
            id: `${chapterId}-${verseNumber}`,
            reference: `Versículo ${verseNumber}`,
            text: verseText,
          });
        }
      }

      setChapterVerses(versesArray);
    } catch (error) {
      console.error("Error fetching chapter content:", error);
    }
  };

  const fetchData = async () => {
    try {
      const response = await getBibles();
      if (response && response.data) {
        setBibleVerse(response.data);
      } else {
        console.error("Unexpected API response structure:", response);
      }
    } catch (error) {
      console.error("Error fetching bibles:", error);
    }
  };

  const handleBook = (bookId: string) => {
    setSelectedBook(bookId);
    toggleSidebar();
  };

  const loadChapters = async (bookId: string) => {
    try {
      const response = await getChapters(bookId);
      if (response && response.data) {
        setBibleVerseChapters(response.data);
      } else {
        console.error("Unexpected API response structure:", response);
      }
    } catch (error) {
      console.error("Error fetching chapters:", error);
    }
  };

  const searchBibleVerse = async (query: string) => {
    try {
      const response = await searchBible(query);
      if (response && response.data) {
        setBibleVerse(response.data);
      } else {
        console.error("Unexpected API response structure:", response);
      }
    } catch (error) {
      console.error("Error fetching bible verse:", error);
    }
  };
  return (
    <BibleContext.Provider
      value={{
        bibleVerse,
        selectedBook,
        bibleVerseChapters,
        chapterVerses,
        handleBook,
        loadChapterVerses,
        scrollAreaRef,
        searchBibleVerse,
        toggleSidebar,
      }}
    >
      {children}
    </BibleContext.Provider>
  );
};

// eslint-disable-next-line react-refresh/only-export-components
export const useBible = (): BibleContextProps => {
  const context = useContext(BibleContext);
  if (!context) {
    throw new Error("useBible must be used within a BibleProvider");
  }
  return context;
};
