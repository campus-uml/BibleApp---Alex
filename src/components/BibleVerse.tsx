import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
import { ScrollArea, ScrollBar } from "@/components/ui/scroll-area";
import { useBible } from "../context/BIbleContext";
import { Copy, Heart, ArrowLeft, Share, Play, Book } from "lucide-react";
import type { SearchResults, Verse } from "@/types";
import { Button } from "@/components/ui/button";
import { useActions } from "@/Hooks/useActions";

interface BibleVerseProps {
  searchResults: SearchResults | null;
  onClearSearch: () => void;
  addFavorite: (verse: Verse) => void;
  removeFavorite: (verseId: string) => void;
  favorites: Verse[];
}

export const BibleVerse = ({
  searchResults,
  onClearSearch,
  addFavorite,
  favorites,
  removeFavorite,
}: BibleVerseProps) => {
  const {
    bibleVerseChapters,
    chapterVerses,
    loadChapterVerses,
    scrollAreaRef,
    selectedBook,
    bibleVerse,
  } = useBible();
  const { copyToClipboard, shareToWhatsApp } = useActions();

  const bookName = (selectedBook: string) => {
    const book = bibleVerse.find((book) => book.id === selectedBook);
    return book ? book.name : "";
  };

  return (
    <div className="min-h-screen bg-gray-50">
      <Tabs
        defaultValue="1"
        className="w-full max-w-[21.2rem] md:max-w-6xl mx-auto"
      >
        <div className="sticky top-0 bg-gray-50 z-10 pb-4">
          {searchResults ? (
            <div className="flex items-center justify-between mb-4">
              <Button
                variant="outline"
                onClick={onClearSearch}
                className="flex items-center gap-2"
              >
                <ArrowLeft size={16} />
                Volver
              </Button>
              <h2 className="text-xl font-semibold text-gray-800">
                Resultados de búsqueda
              </h2>
            </div>
          ) : (
            <>
              <h1 className="text-2xl font-bold text-center text-gray-800 mb-4">
                {bookName(selectedBook)}
              </h1>
              {bibleVerseChapters.length > 0 ? (
                <TabsList className="h-12 items-center justify-start rounded-md bg-muted p-1 text-muted-foreground w-full overflow-hidden">
                  <ScrollArea
                    ref={scrollAreaRef}
                    className="w-full overflow-x-auto whitespace-nowrap scrollbar-thin scrollbar-thumb-gray-300 scrollbar-track-gray-100"
                  >
                    <div className="flex items-center gap-2 py-1">
                      {bibleVerseChapters.map((chapter) => (
                        <TabsTrigger
                          key={chapter.id}
                          value={chapter.number.toString()}
                          onClick={() => loadChapterVerses(chapter.id)}
                          className="ring-offset-background focus-visible:ring-ring data-[state=active]:bg-background data-[state=active]:text-foreground px-3 py-1.5 rounded-md"
                        >
                          {chapter.number}
                        </TabsTrigger>
                      ))}
                    </div>
                    <ScrollBar orientation="horizontal" />
                  </ScrollArea>
                </TabsList>
              ) : (
                <div className="text-center py-8">
                  <Book className="w-12 h-12 text-slate-300 mx-auto mb-3" />
                  <p className="text-lg font-semibold text-slate-600">
                    No hay versículos aún
                  </p>
                  <p className="text-slate-500 mb-4">
                    Selecciona en la parte superior para comenzar
                  </p>
                </div>
              )}
            </>
          )}
        </div>

        {searchResults &&
        searchResults.verses &&
        searchResults.verses.length > 0 ? (
          <div className="space-y-4 mt-4">
            {searchResults.verses.map((verse, index) => (
              <div key={index} className="bg-white shadow-md rounded-lg p-2">
                <h3 className="text-sm text-slate-500 font-semibold mb-2">
                  {verse.id}
                </h3>
                <p className="text-base text-gray-700">{verse.text}</p>
                <div className="flex justify-end gap-4 mt-4">
                  <button
                    onClick={() =>
                      favorites.some((fav) => fav.id === verse.id)
                        ? removeFavorite(verse.id)
                        : addFavorite({ ...verse, reference: verse.id })
                    }
                    className="mt-2 focus:outline-none"
                    aria-label={
                      favorites.some((fav) => fav.id === verse.id)
                        ? "remover"
                        : "agregar"
                    }
                  >
                    <Heart
                      className={`w-6 h-6 ${
                        favorites.some((fav) => fav.id === verse.id)
                          ? "text-red-500"
                          : "text-gray-900"
                      }`}
                    />
                  </button>
                  <button className="text-gray-600 hover:text-blue-500 transition-colors">
                    <Play size={20} />
                  </button>
                  <button
                    onClick={() => copyToClipboard(verse.text)}
                    className="text-gray-600 hover:text-green-500 transition-colors"
                  >
                    <Copy size={20} />
                  </button>
                  <button
                    onClick={() => shareToWhatsApp(verse.text)}
                    className="text-gray-600 hover:text-purple-500 transition-colors"
                  >
                    <Share size={20} />
                  </button>
                </div>
              </div>
            ))}
          </div>
        ) : (
          <div className="space-y-4 mt-4">
            {bibleVerseChapters.map((chapter) => (
              <TabsContent
                key={chapter.id}
                value={chapter.number.toString()}
                className="bg-white rounded-lg shadow-sm p-2"
              >
                <h2 className="text-xl text-center font-semibold mb-4 text-gray-800">
                  Capítulo {chapter.number}
                </h2>
                <div className="space-y-4">
                  {chapterVerses.map((verse, index) => (
                    <div
                      key={index}
                      className="bg-white shadow-md rounded-lg p-4"
                    >
                      <h3 className="text-sm text-slate-500 font-semibold mb-2">
                        {verse.reference}
                      </h3>
                      <p className="text-gray-700 leading-relaxed italic">
                        {verse.text}
                      </p>
                      <div className="flex justify-end gap-4 mt-4">
                        <button
                          onClick={() =>
                            favorites.some((fav) => fav.id === verse.id)
                              ? removeFavorite(verse.id)
                              : addFavorite({ ...verse, reference: verse.id })
                          }
                          className="mt-2 focus:outline-none"
                          aria-label={
                            favorites.some((fav) => fav.id === verse.id)
                              ? "remover"
                              : "agregar"
                          }
                        >
                          <Heart
                            size={20}
                            className={` ${
                              favorites.some((fav) => fav.id === verse.id)
                                ? "text-red-500"
                                : "text-gray-900"
                            }`}
                          />
                        </button>
                        <button className="text-gray-600 hover:text-blue-500 transition-colors">
                          <Play size={20} />
                        </button>
                        <button
                          onClick={() => copyToClipboard(verse.text)}
                          className="text-gray-600 hover:text-green-500 transition-colors"
                        >
                          <Copy size={20} />
                        </button>
                        <button
                          onClick={() => shareToWhatsApp(verse.text)}
                          className="text-gray-600 hover:text-purple-500 transition-colors"
                        >
                          <Share size={20} />
                        </button>
                      </div>
                    </div>
                  ))}
                </div>
              </TabsContent>
            ))}
          </div>
        )}
      </Tabs>
    </div>
  );
};

export default BibleVerse;
