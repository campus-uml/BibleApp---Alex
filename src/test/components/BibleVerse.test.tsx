import { render, screen } from "@testing-library/react";
import { describe, it, expect, vi, Mock } from "vitest";
import BibleVerse from "../../components/BibleVerse"; // Adjust the import path accordingly
import { useBible } from "../../context/BIbleContext";
import { useActions } from "@/Hooks/useActions";

vi.mock("../../context/BIbleContext", () => ({
  useBible: vi.fn().mockReturnValue({
    bibleVerseChapters: [],
    chapterVerses: [],
    loadChapterVerses: vi.fn(),
    scrollAreaRef: null,
    selectedBook: "genesis",
    bibleVerse: [],
  }),
}));
vi.mock("@/Hooks/useActions", () => ({
  useActions: vi.fn(),
}));

describe("Pruebas en BibleVerse", () => {
  it("debería renderizar la lista de versículos de la Biblia cuando hay resultados de búsqueda disponibles", () => {
    const mockSearchResults = {
      verses: [
        {
          id: "1",
          text: "En el principio Dios creó los cielos y la tierra.",
          orgId: "1",
          bookId: "1",
          bibleId: "1",
          chapterId: "1",
        },
        {
          id: "2",
          text: "Y dijo Dios: Sea la luz; y fue la luz.",
          orgId: "2",
          bookId: "1",
          bibleId: "1",
          chapterId: "1",
        },
      ],
      length: 2,
      query: "genesis",
      limit: 10,
      offset: 0,
      total: 2,
      verseCount: 2,
      passages: [],
    };

    const mockClearSearch = vi.fn();
    const mockAddFavorite = vi.fn();
    const mockRemoveFavorite = vi.fn();
    const mockFavorites: {
      id: string;
      reference: string;
      text: string;
      orgId: string;
      bookId: string;
      bibleId: string;
      chapterId: string;
    }[] = [];

    (useBible as Mock).mockReturnValue({
      bibleVerseChapters: [],
      chapterVerses: [],
      loadChapterVerses: vi.fn(),
      scrollAreaRef: null,
      selectedBook: "genesis",
      bibleVerse: [],
    });
    (useActions as Mock).mockReturnValue({
      copyToClipboard: vi.fn(),
      shareToWhatsApp: vi.fn(),
    });

    render(
      <BibleVerse
        searchResults={mockSearchResults}
        onClearSearch={mockClearSearch}
        addFavorite={mockAddFavorite}
        removeFavorite={mockRemoveFavorite}
        favorites={mockFavorites}
      />
    );

    expect(screen.getByText("Resultados de búsqueda")).toBeInTheDocument();
  });

  it("debería renderizar un mensaje cuando no hay resultados de búsqueda disponibles", () => {
    const mockSearchResults = null;
    const mockClearSearch = vi.fn();
    const mockAddFavorite = vi.fn();
    const mockRemoveFavorite = vi.fn();
    const mockFavorites: {
      id: string;
      reference: string;
      text: string;
      orgId: string;
      bookId: string;
      bibleId: string;
      chapterId: string;
    }[] = [];

    (useBible as Mock).mockReturnValue({
      bibleVerseChapters: [],
      chapterVerses: [],
      loadChapterVerses: vi.fn(),
      scrollAreaRef: null,
      selectedBook: "genesis",
      bibleVerse: [],
    });
    (useActions as Mock).mockReturnValue({
      copyToClipboard: vi.fn(),
      shareToWhatsApp: vi.fn(),
    });

    render(
      <BibleVerse
        searchResults={mockSearchResults}
        onClearSearch={mockClearSearch}
        addFavorite={mockAddFavorite}
        removeFavorite={mockRemoveFavorite}
        favorites={mockFavorites}
      />
    );

    expect(screen.getByText("No hay versículos aún")).toBeInTheDocument();
  });
  // it("debería renderizar el nombre del libro y capítulo seleccionado", async () => {
  //   const mockSearchResults = null;
  //   const mockClearSearch = vi.fn();
  //   const mockAddFavorite = vi.fn();
  //   const mockRemoveFavorite = vi.fn();
  //   const mockFavorites: {
  //     id: string;
  //     reference: string;
  //     text: string;
  //     orgId: string;
  //     bookId: string;
  //     bibleId: string;
  //     chapterId: string;
  //   }[] = [];
  
  //   (useBible as Mock).mockReturnValue({
  //     bibleVerseChapters: [],
  //     chapterVerses: [],
  //     loadChapterVerses: vi.fn(),
  //     scrollAreaRef: null,
  //     selectedBook: "genesis", // 'genesis' en minúsculas
  //     selectedChapter: 1,
  //     bibleVerse: [
  //       {
  //         id: "1",
  //         name: "Génesis", // 'Génesis' con mayúscula en la respuesta
  //         abbreviation: "Gn",
  //         chapters: 50,
  //         orgId: "1",
  //         bibleId: "1",
  //       },
  //     ],
  //   });
  
  //   (useActions as Mock).mockReturnValue({
  //     copyToClipboard: vi.fn(),
  //     shareToWhatsApp: vi.fn(),
  //   });
  
  //   render(
  //     <BibleVerse
  //       searchResults={mockSearchResults}
  //       onClearSearch={mockClearSearch}
  //       addFavorite={mockAddFavorite}
  //       removeFavorite={mockRemoveFavorite}
  //       favorites={mockFavorites}
  //     />
  //   );
  
  //   // Usa findByText y espera que el texto esté en el documento
  //   expect(await screen.findByText("Génesis")).toBeInTheDocument();
  //   expect(await screen.findByText("Capítulo 1")).toBeInTheDocument();
  // });
  

  
});
