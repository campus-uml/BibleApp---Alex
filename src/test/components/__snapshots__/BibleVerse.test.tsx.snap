// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en BibleVerse > debería renderizar la lista de versículos de la Biblia cuando hay resultados de búsqueda disponibles 1`] = `
<DocumentFragment>
  <div>
    <div
      class="w-full max-w-[21.2rem] md:max-w-6xl mx-auto"
      data-orientation="horizontal"
      dir="ltr"
    >
      <div
        class="sticky top-0 z-10 pb-4 bg-background"
      >
        <div
          class="flex items-center justify-between mb-4"
        >
          <button
            class="justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground h-9 px-4 py-2 flex items-center gap-2"
          >
            <svg
              class="lucide lucide-arrow-left"
              fill="none"
              height="16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m12 19-7-7 7-7"
              />
              <path
                d="M19 12H5"
              />
            </svg>
            Volver
          </button>
          <h2
            class="text-xl font-semibold"
          >
            Resultados de búsqueda
          </h2>
        </div>
      </div>
      <div
        class="space-y-4 mt-4"
      >
        <div
          class="bg-card text-card-foreground shadow-md rounded-lg p-4"
        >
          <h3
            class="text-sm text-muted-foreground font-semibold mb-2"
          >
            1
          </h3>
          <p
            class="text-base"
          >
            En el principio Dios creó los cielos y la tierra.
          </p>
          <div
            class="flex justify-end gap-4 mt-4"
          >
            <button
              aria-label="agregar"
              class="mt-2 focus:outline-none"
            >
              <svg
                class="lucide lucide-heart w-6 h-6 text-muted-foreground"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
                />
              </svg>
            </button>
            <button
              class="text-muted-foreground hover:text-blue-500 transition-colors"
            >
              <svg
                class="lucide lucide-play"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="6 3 20 12 6 21 6 3"
                />
              </svg>
            </button>
            <button
              class="text-muted-foreground hover:text-green-500 transition-colors"
            >
              <svg
                class="lucide lucide-copy"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="14"
                  rx="2"
                  ry="2"
                  width="14"
                  x="8"
                  y="8"
                />
                <path
                  d="M4 16c-1.1 0-2-.9-2-2V4c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2"
                />
              </svg>
            </button>
            <button
              class="text-muted-foreground hover:text-purple-500 transition-colors"
            >
              <svg
                class="lucide lucide-share"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
                />
                <polyline
                  points="16 6 12 2 8 6"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="2"
                  y2="15"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="bg-card text-card-foreground shadow-md rounded-lg p-4"
        >
          <h3
            class="text-sm text-muted-foreground font-semibold mb-2"
          >
            2
          </h3>
          <p
            class="text-base"
          >
            Y dijo Dios: Sea la luz; y fue la luz.
          </p>
          <div
            class="flex justify-end gap-4 mt-4"
          >
            <button
              aria-label="agregar"
              class="mt-2 focus:outline-none"
            >
              <svg
                class="lucide lucide-heart w-6 h-6 text-muted-foreground"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 14c1.49-1.46 3-3.21 3-5.5A5.5 5.5 0 0 0 16.5 3c-1.76 0-3 .5-4.5 2-1.5-1.5-2.74-2-4.5-2A5.5 5.5 0 0 0 2 8.5c0 2.3 1.5 4.05 3 5.5l7 7Z"
                />
              </svg>
            </button>
            <button
              class="text-muted-foreground hover:text-blue-500 transition-colors"
            >
              <svg
                class="lucide lucide-play"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="6 3 20 12 6 21 6 3"
                />
              </svg>
            </button>
            <button
              class="text-muted-foreground hover:text-green-500 transition-colors"
            >
              <svg
                class="lucide lucide-copy"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="14"
                  rx="2"
                  ry="2"
                  width="14"
                  x="8"
                  y="8"
                />
                <path
                  d="M4 16c-1.1 0-2-.9-2-2V4c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2"
                />
              </svg>
            </button>
            <button
              class="text-muted-foreground hover:text-purple-500 transition-colors"
            >
              <svg
                class="lucide lucide-share"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
                />
                <polyline
                  points="16 6 12 2 8 6"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="2"
                  y2="15"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Pruebas en BibleVerse > debería renderizar un mensaje cuando no hay resultados de búsqueda disponibles 1`] = `
<DocumentFragment>
  <div>
    <div
      class="w-full max-w-[21.2rem] md:max-w-6xl mx-auto"
      data-orientation="horizontal"
      dir="ltr"
    >
      <div
        class="sticky top-0 z-10 pb-4 bg-background"
      >
        <h1
          class="text-2xl font-bold text-center mb-4"
        />
        <div
          class="text-center py-8"
        >
          <svg
            class="lucide lucide-book w-12 h-12 text-muted-foreground mx-auto mb-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 19.5v-15A2.5 2.5 0 0 1 6.5 2H19a1 1 0 0 1 1 1v18a1 1 0 0 1-1 1H6.5a1 1 0 0 1 0-5H20"
            />
          </svg>
          <p
            class="text-lg font-semibold"
          >
            No hay versículos aún
          </p>
          <p
            class="text-muted-foreground mb-4"
          >
            Selecciona en la parte superior para comenzar
          </p>
        </div>
      </div>
      <div
        class="space-y-4 mt-4"
      />
    </div>
  </div>
</DocumentFragment>
`;
